src/AnalysisPrograms/Recognizers/PteropusSpecies.cs

```diff
@@ -0,0 +1,450 @@
+    using AudioAnalysisTools;
+    using AudioAnalysisTools.DSP;
+    using AudioAnalysisTools.Indices;
+    using AudioAnalysisTools.StandardSpectrograms;
+    using AudioAnalysisTools.WavTools;
+    using log4net;
+    using TowseyLibrary;
+
+    /// <summary>
+    /// This is a recognizer for species of Flying Fox, Pteropus species.
+    /// </summary>
+    internal class PteropusSpecies : RecognizerBase
+    {
+        public override string Author => "Towsey";
+
+        public override string SpeciesName => "PteropusSpecies";
+
+        public override string Description => "[STATUS DESCRIPTION] Detects acoustic events for species of Flying Fox, Pteropus species";
+
+        private static readonly ILog log = LogManager.GetLogger(MethodBase.GetCurrentMethod().DeclaringType);
+
+        /// <summary>
+        /// Summarize your results. This method is invoked exactly once per original file.
+        /// </summary>
+        public override void SummariseResults(
+            AnalysisSettings settings,
+            FileSegment inputFileSegment,
+            EventBase[] events,
+            SummaryIndexBase[] indices,
+            SpectralIndexBase[] spectralIndices,
+            AnalysisResult2[] results)
+        {
+            // No operation - do nothing. Feel free to add your own logic.
+            base.SummariseResults(settings, inputFileSegment, events, indices, spectralIndices, results);
+        }
+
+        /// <summary>
+        /// This method is called once per segment (typically one-minute segments).
+        /// </summary>
+        /// <param name="audioRecording">one minute of audio recording.</param>
+        /// <param name="genericConfig">config file that contains parameters used by all profiles.</param>
+        /// <param name="segmentStartOffset">when recording starts.</param>
+        /// <param name="getSpectralIndexes">not sure what this is.</param>
+        /// <param name="outputDirectory">where the recogniser results can be found.</param>
+        /// <param name="imageWidth"> assuming ????.</param>
+        /// <returns>recogniser results.</returns>
+        public override RecognizerResults Recognize(AudioRecording audioRecording, Config genericConfig, TimeSpan segmentStartOffset, Lazy<IndexCalculateResult[]> getSpectralIndexes, DirectoryInfo outputDirectory, int? imageWidth)
+        {
+            // Examples of the APIs available. You don't need all of these commands! Pick and choose.
+            if (ConfigFile.HasProfiles(genericConfig))
+            {
+                string[] profileNames = ConfigFile.GetProfileNames<Config>(genericConfig);
+                int count = profileNames.Length;
+                var message = new StringBuilder("Found " + count + " config profile(s): ");
+                foreach (string s in profileNames)
+                {
+                    message.Append(s + ", ");
+                }
+
+                log.Info(message.ToString());
```

Comment:
this is probably better as a `.Debug` or `.Verbose` logging message - we don't need to see this every time a one-minute segment is analyzed